x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/RequestXContent.java

```diff
@@ -155,52 +167,67 @@ private static QueryParams parseParams(XContentParser p) throws IOException {
                         );
                     }
                     for (Map.Entry<String, Object> entry : param.fields.entrySet()) {
-                        String name = entry.getKey();
-                        if (isValidParamName(name) == false) {
-                            errors.add(
-                                new XContentParseException(
-                                    loc,
-                                    "["
-                                        + name
-                                        + "] is not a valid parameter name, "
-                                        + "a valid parameter name starts with a letter or underscore, "
-                                        + "and contains letters, digits and underscores only"
-                                )
-                            );
+                        ParserUtils.ParamClassification classification;
+                        String paramName = entry.getKey();
+                        checkParamNameValidity(paramName, errors, loc);
+
+                        if (entry.getValue() instanceof HashMap<?, ?> values) {// parameter specified as key:value pairs
+                            HashMap<ParamParsingKeys, Object> paramElements = new HashMap<>(3);
```

Comment:
Map<> paramElements = ..
Separately the initial capacity can't be 3 since it needs to take into account the load factor. Use `Maps.newMapWithExpectedSize` (will create a HashMap behind the scenes).